app/models/bulkrax/entry.rb

```diff
@@ -3,16 +3,20 @@ class Entry < ApplicationRecord
     include Bulkrax::Concerns::HasMatchers
     include Bulkrax::Concerns::HasLocalProcessing
 
-    belongs_to :importer
+    # @deprecated Please use importerexporter instead
+    #custom_deprecator = ActiveSupport::Deprecation.new('next-release', 'bulkrax')
+    #ActiveSupport::Deprecation.deprecate_methods(Bulkrax::Entry, importer: :importerexporter, deprecator: custom_deprecator)
+    belongs_to :importer, required: false
+    
+    belongs_to :importerexporter, polymorphic: true
     serialize :parsed_metadata, JSON
     # do not serialize raw_metadata as so we can support xml or other formats
     # keep it raw.
     serialize :collection_ids, Array
 
     attr_accessor :all_attrs, :last_exception
 
-    delegate :parser, :mapping,
-             to: :importer
+    delegate :parser, :mapping, to: :importerexporter
```

Comment:
if you are making this change, you are not really deprecating the above, you are just removing it.  I think that's ok, just worth pointing out.  What you may want to do is remove belongs_to :importer above but put it back in the migration by opening up the class in the migration file and adding it.

Reply:
ahh, I didn't think of doing it like that, a much better way, although it was fun figuring out how to do the deprecation messages.

Reply:
I'll merge and do a small additional PR